tasks/js.js

```
@@ -92,8 +92,8 @@ gulp.task('js-browserify', done => {
 
     const tasks = files
       .map(bundleFile)
-      .map(bundler => new Promise(resolve => bundler.on('end', resolve)))
+      .map(bundle => new Promise(resolve => bundle.on('finish', resolve)))
 
-    Promise.all(tasks).then(done)
+    Promise.all(tasks).then(() => done()).catch(done)
```

Maybe I'm mistaking, but it looks like `Promise.all` won't catch any errors with the bundle since the rejection's not handled. I'm guessing you added it so that the Gulp _js_ task finishes instead of hanging or timing out, but right now I believe it's going to fail silently. What do you think about adding a `bundler.on('error'...)` handler?

Also, the `done` callback didn't work before, maybe it's safer to call it with an arrow function `.catch(() => done(error))` instead.

---

Hi Raz, 

The catch() callback is for any exceptions outside of browserify (which I cannot think of right now), as an extra safe guard. E.g., you add a plugin in a wrong way, something like that. bundler.on('error'...) is handled by the errorify plugin.

The issue was caused because in the previous case, the Promise.all would resolve with an array. If you pass the done callback anything it thinks that's an error, which is why the task failed. I guess it didn't show anything in the console, because done() wasn't called with a real Error object.